examples/diff.c

```diff
@@ -290,9 +305,22 @@ static void parse_opts(struct opts *o, int argc, char *argv[])
 			o->findopts.flags |= GIT_DIFF_FIND_COPIES;
 		else if (!strcmp(a, "--find-copies-harder"))
 			o->findopts.flags |= GIT_DIFF_FIND_COPIES_FROM_UNMODIFIED;
-		else if (is_prefixed(a, "-B") || is_prefixed(a, "--break-rewrites"))
-			/* TODO: parse thresholds */
+		else if (optional_str_arg(
+				break_rewrite_arg, &args, "-B", NULL) || 
+			optional_str_arg(
+				break_rewrite_arg, &args, "--break-rewrites", NULL)) {
+
 			o->findopts.flags |= GIT_DIFF_FIND_REWRITES;
+			err = extract_break_rewrites(&m, &n, *break_rewrite_arg);
+			if (err==-1) 
```

Comment:
This goes according to the standards, but here this is for demonstration purpose (examples/diff.c) and user should know the reason why it failed and program should terminate. We need to just provide a functionality to the user, how to use the library, after that it's his choice how to use.